pkg/client/client.go

```diff
@@ -37,66 +36,139 @@ import (
 
 const (
 	defaultEndpointPort = "9440"
-	ProviderName        = "nutanix"
-	configPath          = "/etc/nutanix/config"
-	endpointKey         = "prismCentral"
-	capxNamespaceKey    = "POD_NAMESPACE"
+
+	capxNamespaceKey = "POD_NAMESPACE"
+)
+
+var (
+	// use a var to override in tests
+	configPath = "/etc/nutanix/config/prismCentral"
 )
 
-var ErrPrismAddressNotSet = fmt.Errorf("cannot get credentials if Prism Address is not set")
-var ErrPrismPortNotSet = fmt.Errorf("cannot get credentials if Prism Port is not set")
+var (
+	ErrPrismAddressNotSet = fmt.Errorf("cannot get credentials if Prism Address is not set")
+	ErrPrismPortNotSet    = fmt.Errorf("cannot get credentials if Prism Port is not set")
+
+	ErrPrismIUsernameNotSet = fmt.Errorf("could not create client because username was not set")
+	ErrPrismIPasswordNotSet = fmt.Errorf("could not create client because password was not set")
+
+	ErrCredentialRefNotSet = fmt.Errorf("credentialRef must be set on CAPX manager")
+)
 
 type NutanixClientHelper struct {
 	secretInformer    coreinformers.SecretInformer
 	configMapInformer coreinformers.ConfigMapInformer
+
+	managerNutanixPrismEndpointReader func() (*credentials.NutanixPrismEndpoint, error)
 }
 
-func NewNutanixClientHelper(secretInformer coreinformers.SecretInformer, cmInformer coreinformers.ConfigMapInformer) (*NutanixClientHelper, error) {
+func NewHelper(secretInformer coreinformers.SecretInformer, cmInformer coreinformers.ConfigMapInformer) *NutanixClientHelper {
 	return &NutanixClientHelper{
-		secretInformer:    secretInformer,
-		configMapInformer: cmInformer,
-	}, nil
+		secretInformer:                    secretInformer,
+		configMapInformer:                 cmInformer,
+		managerNutanixPrismEndpointReader: readManagerNutanixPrismEndpointFromDefaultFile,
```

Comment:
this should also be something we should be able to pass into the constructor

Reply:
Previously this package held the constant `configPath` and knew how to read that file.
I agree, thats probably more of a concern of the controller, I'll pull this out as you suggested, thanks!

Reply:
Actually after making these changes I don't think it makes much sense to take this in in the construction either.
At some point we should refactor this more and have the API similar to other providers' `identityRef`, e.g. https://cluster-api-aws.sigs.k8s.io/topics/multitenancy.

What do you think keeping it the way it is now with a private decorator thats used in the unit tests, and then working on a further API changes in the future?

Reply:
@dkoshkin fine by me.